### PR TITLE
Add support for ticky definition json fields

### PR DESCRIPTION
**Pull request by AndreasPK**
See https://gitlab.haskell.org/ghc/ghc/-/merge_requests/8514

### CODE REVIEW

**Comment by Mikolaj** [comment]
Oh, so that's why `eventlog2html` was crashing for me when I tried to feed it HEAD's evenlog from ticky? Thanks a lot for the PR!

**Comment by AndreasPK** [comment]
> Oh, so that's why `eventlog2html` was crashing for me when I tried to feed it HEAD's evenlog from ticky? Thanks a lot for the PR!

Matt at least thinks it's because https://github.com/haskell/ghc-events/pull/83 hasn't been merged yet.

File: src/GHC/RTS/Events/Binary.hs [hunk 1]
```diff
@@ -921,12 +921,14 @@ binaryEventParsers =
     return $! UserBinaryMessage { payload }
   ]
 
+-- | Reads the `a` object, and if that didn't consume the complete
+-- event skip over the leftover data.
```
Comment (maoe) [inline src/GHC/RTS/Events/Binary.hs:925]: Nice to see the comment here.

File: src/GHC/RTS/Events/Binary.hs [hunk 2]
```diff
@@ -936,6 +938,9 @@ skipExtra expected get_body = do
 
 -- | Skip extra stuff at the end of a variable sized event
 -- For forwards compatability (allowing newer events to be read with older versions)
+--
+-- The body parser expectes an expected length for the input (stored in the eventlog)
+-- a position of the parser
```
Comment (maoe) [inline src/GHC/RTS/Events/Binary.hs:943]: Maybe this Haddock comment should be attached to the type signature of the body_parser.

**Review by maoe — COMMENTED** [review]
682c53d39658cba7e58843e507e9a8e1518669b1 looks good.

File: src/GHC/RTS/Events/Binary.hs (hunk 2 above)
Comment (AndreasPK) [reply src/GHC/RTS/Events/Binary.hs]: If I understand you correctly you mean something like:

```
-- | Skip extra stuff at the end of a variable sized event
-- For forwards compatability (allowing newer events to be read with older versions)
variableSizeParser :: Int
                   -> (Word16 -> Int -> Get a) -- The body parser expectes an expected length for the input (stored in the eventlog)
                                               -- a position of the parser
                   -> EventParser a
```

But then the comment won't show up in the actual haddocks anymore I think?

So I would rather keep it part of the proper haddocks. I did rephrase the comment slightly as after not looking at it for half a year it wasn't as clear as it seemed back then.

**Comment by AndreasPK** [comment]
@maoe Let me know if anything more is required to get this merged.

**Comment by AndreasPK** [comment]
Rebased on https://github.com/haskell/ghc-events/pull/83.

File: src/GHC/RTS/Events/Binary.hs (hunk 2 above)
Comment (AndreasPK) [reply src/GHC/RTS/Events/Binary.hs]: Seems matt went ahead and made this change in https://github.com/haskell/ghc-events/pull/83.

**Comment by maoe** [comment]
Thanks I'll take a look in a few days.

**Comment by maoe** [comment]
LGTM. Thanks!

**Pull request merged**